lib/email_alert_title_builder.rb

```diff
@@ -0,0 +1,68 @@
+class EmailAlertTitleBuilder
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(filter:, subscription_list_title_prefix:, facets:)
+    @filter = filter
+    @subscription_list_title_prefix = subscription_list_title_prefix
+    @facets = facets
+  end
+
+  def call
+    if facets.empty?
+      prefix
+    elsif facets.size == 1
+      (prefix + topic_names_sentence(facets.first)).upcase_first
+    elsif selected_facets.empty?
+      prefix
```

Comment:
feels like we are calling `prefix` a lot. I'm not sure it's a problem, but I wonder if the code might read a little more elegantly if we pulled out the conditional logic into a method.

Every case here uses `prefix` so something like:

```
def call
  prefix + suffix
end

def suffix
  if facets.size == 1
      topic_names_sentence(facets.first)).upcase_first
  elsif facets.size > 1
    facet_sentences = selected_facets.map do |facet|
      facet["facet_name"] + " of " + topic_names_sentence(facet)
    end
    facet_sentences.to_sentence).upcase_first
  end
end
```

I'm not sure if I've captured the logic there though? i.e. is that catch-all `elsif` always captured by the case of `facets.size > 1`